runtime/src/rent_collector.rs

```diff
@@ -52,6 +58,36 @@ impl RentCollector {
         }
     }
 
+    /// true if it is easy to determine this account should not have rent collected from it
+    pub fn no_rent(
+        &self,
+        address: &Pubkey,
+        account: &impl ReadableAccount,
+        rent_for_sysvars: bool,
+    ) -> bool {
+        account.executable() // executable accounts must be rent-exempt balance
+            || account.rent_epoch() > self.epoch
+            || (!rent_for_sysvars && sysvar::check_id(account.owner()))
+            || *address == incinerator::id()
+    }
+
+    /// true if it is easy to determine this account should not have rent collected from it
```

Comment:
Minor: Looks like this comment is for the other function maybe?

Reply:
fixed. good catch.